src/ecrc-frontend/src/components/page/consent/Consent.js

```diff
@@ -306,70 +311,111 @@ export default function Consent({
       .then(getSessionResponse => {
         sessionId = getSessionResponse.data.sessionId;
 
-        const newCRC = {
+        appInfo = {
+          ...appInfo,
+          sessionId
+        };
+
+        CRC = {
           ...CRC,
           applPartyId: partyId,
           sessionId
         };
 
         if (orgApplicantRelationship === "ONETIME") {
-          newCRC.orgApplToPay = "O";
+          CRC.orgApplToPay = "O";
         } else if (orgApplicantRelationship === "EMPLOYEE") {
-          newCRC.orgApplToPay = "A";
+          CRC.orgApplToPay = "A";
         }
 
-        return axios.post("/ecrc/private/createNewCRCService", newCRC, {
+        if (orgApplicantRelationship === "VOLUNTEER") {
+          axios
+            .post("/ecrc/private/createNewCRCService", CRC, {
+              headers: {
+                Authorization: `Bearer ${token}`
+              }
+            })
+            .then(volunteerCRCResponse => {
+              serviceId = volunteerCRCResponse.data.serviceId;
+
+              appInfo = {
+                ...appInfo,
+                serviceId
+              };
+
+              setApplicationInfo(appInfo);
+
+              toSuccess();
+              setLoading(false);
+            });
+
+          return false;
+        }
+
+        return axios.get(
+          `/ecrc/private/getNextInvoiceId?orgTicketNumber=${orgTicketNumber}&requestGuid=${uuid}`,
+          {
+            headers: {
+              Authorization: `Bearer ${token}`
+            }
+          }
+        );
+      })
+      .then(getInvoiceResponse => {
+        if (!getInvoiceResponse) {
+          return getInvoiceResponse;
+        }
+
+        invoiceId = getInvoiceResponse.data.invoiceId;
+
+        appInfo = {
+          ...appInfo,
+          invoiceId
+        };
+
+        CRC.invoiceId = invoiceId;
+
+        return axios.post("/ecrc/private/createNewCRCService", CRC, {
           headers: {
             Authorization: `Bearer ${token}`
           }
         });
       })
       .then(newCRCRespose => {
```

Comment:
```suggestion
      .then(newCRCResponse => {
```
and everywhere else :)

Reply:
Fixed